function-controller/src/main/java/io/projectriff/controller/function/FunctionMonitor.java

```diff
@@ -234,20 +242,32 @@ private void scaleFunctions() {
 							logger.debug(
 									"Want {} for {}. Rounded to {} [target = {}]. (Deployment currently set to {})",
 									interpolation, name, rounded, desired, current);
-							if (rounded != current) {
+							if (current == 0 && desired > 0) {
+								// Special case when scaling from 0
+								deployer.deploy(f, 1);
+							}
+							else if (rounded != current) {
 								// Special case when going back to 0
 								if (rounded == 0) {
 									Long start = scaleDownStartTimes.get(name);
 									long now = System.currentTimeMillis();
 									if (start == null) {
 										start = now;
 										scaleDownStartTimes.put(name, now);
+										scaleDownPositionSums.put(name, currentPositionSum);
 									} else {
 										if (now >= start + idleTimeout) {
 											scaleDownStartTimes.remove(name);
 											deployer.deploy(f, rounded);
 										} else {
-											logger.debug("Waiting another {}ms to scale back down to 0 for {}", start + idleTimeout - now, name);
+											if (currentPositionSum > scaleDownPositionSums.get(name)) {
```

Comment:
Why do we need another metric to track "activity" here? "There is still activity" <==> `desired > 0`, isn't it?

Reply:
`desired` is based on *lag* and so it remains at 0 as long as the 1 current replica is keeping up with the events, but we need to reset the clock as long as that 1 current replica is not truly idle

watching for a change in the actual position on any partition is the only option that came to my mind

Reply:
Gotcha